src/core/utils/urlutils.js

```diff
@@ -42,18 +42,15 @@ export function getAnalyticsUrl (env = PRODUCTION, conversionTrackingEnabled = f
 }
 
 /**
- * Returns the passed in url, with all url params from the current url, as well as any
- * pasased in params, appended to it.
+ * Returns the passed in url with the passed in params appended as query params
+ * Note: query parameters in the url are stripped, you should include those query parameters
+ * in `params` if you want to keep them
  * @param {string} url
- * @param {Object} params
+ * @param {SearchParams} params to add to the url
  * @returns {string}
  */
-export function addParamsToUrl (url, params = {}) {
-  const urlParams = new SearchParams(window.location.search.substring(1));
-  for (const paramKey in params) {
-    urlParams.set(paramKey, params[paramKey]);
-  }
-  return url.split('?')[0] + '?' + urlParams;
+export function addParamsToUrl (url, params = new SearchParams()) {
```

Comment:
Since this strips existing query params and so isn't purely additive anymore, do you think the function name should change?

Reply:
was also previously stripping existing query params. can rename to `combineParamsWithUrl` how does that sound?

Reply:
changed to replaceUrlParams at your suggestion